tket/src/Transformations/CliffordResynthesis.cpp

```diff
@@ -49,18 +54,46 @@ static Circuit resynthesised_circuit(
   }
 }
 
+// Returns the two-qubit-gate reduction resulting from resynthesising the given
+// (convex, connected) vertex set using the given transform.
+static int n_2q_reduction(
+    const Circuit &circ, const VertexSet &verts,
+    std::optional<std::function<Circuit(const Circuit &)>> transform,
+    bool allow_swaps) {
+  Subcircuit subcircuit = circ.make_subcircuit(verts);
+  const Circuit subc = circ.subcircuit(subcircuit);
+  Circuit newsubc = resynthesised_circuit(subc, transform, allow_swaps);
+  return int(subc.count_n_qubit_gates(2)) - int(newsubc.count_n_qubit_gates(2));
+}
+
 static bool resynthesise_cliffords(
     Circuit &circ,
     std::optional<std::function<Circuit(const Circuit &)>> transform = nullptr,
     bool allow_swaps = true) {
   bool changed = false;
-  for (const VertexSet &verts :
-       circ.get_subcircuits([](Op_ptr op) { return op->is_clifford(); })) {
+  while (true) {
```

Comment:
Should this maybe better contain the break condition in the while line?

Reply:
It's a matter of preference. That would mean introducing an extra boolean variable outside the loop to control it, and modifying the variable inside the loop, whereas this way we don't need to do that. On balance I think the function is sufficiently small that the logic is apparent.